src/components/widgets/Clock.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+import "../../styles/widget.css"
+import moment from 'moment'
+
+class Clock extends Component {
+
+  constructor( props ) {
+    super( props )
+    this.state = {
+      time: null,
+      amPm: null,
+      dayOfWeek: null,
+      date: null
+    }
+
+    this.setDateAndTime = this.setDateAndTime.bind(this)
+  }
+
+  setDateAndTime() {
+    let currentTime = moment().format( 'h:mm' )
+    let currentAmPm = moment().format( 'A' )
+    let currentDayOfWeek = moment().format( 'dddd' )
+    let currentDate = moment().format( 'D MMM YYYY' )
+
+    this.setState({
+      time: currentTime,
+      amPm: currentAmPm,
+      dayOfWeek: currentDayOfWeek,
+      date: currentDate
+    })
+  }
+
+  componentWillMount() {
+    this.setDateAndTime()
+  }
+
+  componentDidMount() {
+    window.setInterval( function () {
+      this.setDateAndTime()
+    }.bind( this ), 10000)
+  }
+
+  render() {
+    const { specification } = this.props
+
+    const positionStyle = { top: specification.position.x, left: specification.position.y }
```

Comment:
This hints at a leaky abstraction in our domain - the component(s) should not need to know what a specification is; they should only be receiving the data they need to correctly render themselves.  I think the parent to this component should just be supplying a `position` prop, or individual `x` and `y` props.  (See [this](https://en.wikipedia.org/wiki/Law_of_Demeter) for more info.)